test-core/src/main/java/io/micronaut/test/condition/TestActiveCondition.java

```diff
@@ -63,7 +63,11 @@ public boolean matches(ConditionContext context) {
                             final String rootName = packageName + ".$" + simpleName;
                             return declaringTypeClass.isAssignableFrom(activeSpecClazz) || declaringTypeName.equals(rootName) || declaringTypeName.startsWith(rootName + "$");
                         } else {
-                            return declaringTypeClass.isAssignableFrom(activeSpecClazz) || activeSpecName.equals(declaringTypeName) || declaringTypeName.startsWith(activeSpecName + "$");
+                            final String simpleName = NameUtils.getSimpleName(activeSpecName);
+                            return declaringTypeClass.isAssignableFrom(activeSpecClazz) &&
+                                    definition.getClass().getSimpleName().contains(simpleName) // Check if super class def is part of the current spec
```

Comment:
How would this work if you had overlapping names like `SimpleService` and `SimpleServiceHelper` or something?

Reply:
You are right, I have changed the check.